apm-agent-plugins/apm-jms-plugin/src/main/java/co/elastic/apm/agent/jms/JmsInstrumentationHelperImpl.java

```diff
@@ -110,20 +137,74 @@ public void onMessage(Message message) {
         }
     }
 
+    @Nullable
+    public String extractDestinationName(@Nullable Message message, Destination destination) {
+        String destinationName = null;
+        if (message != null) {
+            try {
+                destinationName = message.getStringProperty(JMS_DESTINATION_NAME_PROPERTY);
+            } catch (JMSException e) {
+                logger.warn("Failed to get destination name from message property", e);
+            }
+        }
+        if (destinationName == null) {
+            try {
+                if (destination instanceof Queue) {
+                    destinationName = ((Queue) destination).getQueueName();
+                } else if (destination instanceof Topic) {
+                    destinationName = ((Topic) destination).getTopicName();
+                }
+            } catch (JMSException e) {
+                logger.error("Failed to obtain destination name", e);
+            }
+        }
+        return destinationName;
+    }
+
+    private boolean isTempDestination(Destination destination, @Nullable String extractedDestinationName) {
```

Comment:
Could you describe some high-level concepts around temp queues in `co/elastic/apm/agent/jms/package-info.java`? For example, what are temp queues used for, are they only used in specific implementations ( is `$TEMP$` a convention?),  and why do we need to treat them specially?

Reply:
I will add to the code. In general, those are short-lived ad-hoc queues, for example- a reply queue. This is why they require different handling. Non-temp queues are globally known, very stable entities, like databases. You want those in your service map, but tmp queues will cause a disaster.